### PR TITLE
[CN-1083]: Add doc for container resources

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

**Review by cagric0 — APPROVED** [review]

File: docs/modules/ROOT/pages/resource-configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+= Configuring Container Resources
+
+:description: pass:q[You can configure resources for Hazelcast, sidecar agent and management center containers.]
+
+{description}
+
+Detailed documentation about resources can be found link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[in Resource Management for Pods and Containers documentation of Kubernetes].
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/resource-configuration.adoc:7]:
```suggestion
Detailed documentation about resources can be found in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.
```

File: docs/modules/ROOT/pages/resource-configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+== Configuring Resources for Hazelcast
+
+You can configure resources for Hazelcast containers using the `resources` field of the CRD spec.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/resource-configuration.adoc:11]:
```suggestion
You can configure resources for Hazelcast containers using the `resources` field of the CRD specification.
```

File: docs/modules/ROOT/pages/resource-configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+.Example configuration
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/hazelcast-resources.yaml[]
+----
+
+== Configuring Resources for Sidecar
+
+You can configure resources for sidecar containers using the `agent.resources` field of the CRD spec.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/resource-configuration.adoc:21]:
```suggestion
You can configure resources for sidecar containers using the `agent.resources` field of the CRD specification.
```

File: docs/modules/ROOT/pages/resource-configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+.Example configuration
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/sidecar-resources.yaml[]
+----
+
+== Configuring Resources for Management Center
+
+You can configure resources for Management Center containers using the `resources` field of the CRD spec.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/resource-configuration.adoc:31]:
```suggestion
You can configure resources for Management Center containers using the `resources` field of the CRD specification.
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]
Minor comments (3 or the 4 are simply changing "spec" to "specification")

File: docs/modules/ROOT/pages/resource-configuration.adoc (hunk 2 above)
Comment (kutluhanmetin) [reply docs/modules/ROOT/pages/resource-configuration.adoc:11]: `spec` is a field, so it is better to mention as `spec` IMO.

Comment (rebekah-lawrence) [reply docs/modules/ROOT/pages/resource-configuration.adoc:11]: OK, could we say "the CRD `spec` field" just so it is clear? 

Although that is two "fields" in close proximity....are they both called fields? Could we call `resources` a sub-field? My knowledge of K8s terminology is not good. 

Alternatively, just put the ticks around spec (/`spec/`) - that is probably sufficient.

Comment (hasancelik) [reply docs/modules/ROOT/pages/resource-configuration.adoc]: > Although that is two "fields" in close proximity....are they both called fields? Could we call resources a sub-field?

`resources` is located under `spec`, so it is a kind of sub-field.

> Alternatively, just put the ticks around the spec.a That is probably sufficient.

Done 👍

**Review by hasancelik — APPROVED** [review]

File: docs/modules/ROOT/pages/resource-configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,37 @@
+= Configuring Container Resources
+
+:description: pass:q[You can configure resources for Hazelcast, sidecar agent and management center containers.]
+
+{description}
+
+Detailed documentation about resources can be found in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.
```
Comment (Serdaro) [inline docs/modules/ROOT/pages/resource-configuration.adoc:7]:
```suggestion
You can find detailed information about resources in the link:https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/[Resource Management for Pods and Containers] Kubernetes documentation.
```

**Review by Serdaro — COMMENTED** [review]
Minor suggestions.

**Review by Serdaro — APPROVED** [review]

**Review by rebekah-lawrence — APPROVED** [review]

**Pull request merged**